### ISSUE TITLE
pdf generation resulted in fatal error if file name contain .

### ISSUE THREAD

**Issue by RJerry** [michaelmarty/UniDec#33]
First of all thank you so much for making this wonderful tool that easy to use!!!

I am having Errors when export as pdf report.

Filename: `B0.4_7-9min_CombinedScan_csv`

Error Code from latexpdf:
`! LaTeX Error: Unknown graphics extension: .4_7-9min_CombinedScan_Figure1.pdf.`

I suspect this might due to some string handling issue during the file name process?

Entire Debug info attached:

```
Required multiplierz data file unimod.sqlite not found!  Creating it.
WARNING: No unimod.sqlite found in C:\Users\Ruisi\Desktop\UniDec_Windows_Dist (C:\Users\Ruisi\Desktop\UniDec_Windows_Dist\GUI_UniDec.exe)
Failed to load dynlib/dll 'C:\\Users\\Ruisi\\Desktop\\UniDec_Windows_Dist\\unidec_modules\\waters_importer\\MassLynxRaw.dll'. Most probably this dynlib/dll was not found when the application was frozen.
C:\Users\Ruisi\Desktop\UniDec_Windows_Dist\unidec_modules\waters_importer\MassLynxRaw.dll
False

UniDec Engine v.4.0.0 Beta

UniDec Path: C:\Users\Ruisi\Desktop\UniDec_Windows_Dist\UniDec.exe
Launching UniDec

UniDec Engine v.4.0.0 Beta

UniDec Path: C:\Users\Ruisi\Desktop\UniDec_Windows_Dist\UniDec.exe
Display Size  (2880, 1620)
Loaded:  C:\Users\Ruisi\Desktop\UniDec_Windows_Dist\default_conf.dat
Opening:  B0.4_7-9min_CombinedScan.csv
Opening File:  B0.4_7-9min_CombinedScan.csv
Loading Time: 0.034s
Linear False
Data Prep Done. Time: 0.016s

Read from file: imflag input output numit endz startz zzsig psig beta mzsig psfun massub masslb msig molig massbins mtabsig intthresh aggressive rawflag adductmass nativezub nativezlb poolflag peakshapeinflate noiseflag linflag isotopemode baselineflag orbimode filterwidth zerolog

infile = B0.4_7-9min_CombinedScan_input.dat
outfile = B0.4_7-9min_CombinedScan

UniDec Run
Length of Data: 1471
Threshold: 2.165771
maxlength 4
mzdist set: 0.000000
0.606531
1.000000
0.606531
Number of Oligomers blurred: 3
Number of Charges blurred: 3
Isotope Mode: 0
Iterating

Iteration: 0 Convergence: 0.071958
Iteration: 1 Convergence: 0.022094
Iteration: 10 Convergence: 0.009222
Iteration: 11 Convergence: 0.000016
Iteration: 20 Convergence: 0.000292
Iteration: 21 Convergence: 0.000000
Iteration: 30 Convergence: 0.000006
Iteration: 31 Convergence: 0.000000

Converged in 31 iterations.

Fit data written to: B0.4_7-9min_CombinedScan_fitdat.bin
m/z grid written to: B0.4_7-9min_CombinedScan_grid.bin
Massmax: 40028.320789  Massmin: 24982.508064  Mass axis length: 15045
Mass Grid Written: B0.4_7-9min_CombinedScan_massgrid.bin
Masses written: B0.4_7-9min_CombinedScan_mass.txt
Stats and Error written to: B0.4_7-9min_CombinedScan_error.txt

Error: 0.008255

Finished with 31 iterations and 0.000000 convergence in 0.103000 seconds!

UniDec run 0.12s
File Name:  B0.4_7-9min_CombinedScan.csv R Sqaured:  0.9993822090569541
Plot 3: 0.13s
Plot 5: 0.2s
Plot 1: 0.22s
Plot 2: 0.085s
Peak Picking
Apex: 31090.0 Centroid: 31107.59371793942 FWHM Range: [31080.0, 31139.0]
Apex: 31109.0 Centroid: 31108.70248743998 FWHM Range: [31077.0, 31145.0]
Apex: 31129.0 Centroid: 31109.331674182355 FWHM Range: [31076.0, 31148.0]
Apex: 31823.0 Centroid: 31823.34148653972 FWHM Range: [31815.0, 31832.0]
Apex: 31859.0 Centroid: 31857.998296145535 FWHM Range: [31845.0, 31870.0]
Apex: 31897.0 Centroid: 31853.2739476025 FWHM Range: [31811.0, 31909.0]
Apex: 31930.0 Centroid: 31930.067552155604 FWHM Range: [31916.0, 31944.0]
Apex: 31991.0 Centroid: 32022.471294932457 FWHM Range: [31982.0, 32071.0]
Apex: 32009.0 Centroid: 32023.325776679834 FWHM Range: [31981.0, 32074.0]
Apex: 32031.0 Centroid: 32023.687400030783 FWHM Range: [31981.0, 32075.0]
Apex: 32065.0 Centroid: 32036.943094102644 FWHM Range: [31979.0, 32117.0]
Apex: 32096.0 Centroid: 31953.4425736396 FWHM Range: [31810.0, 32144.0]
Apex: 32163.0 Centroid: 32182.973714408676 FWHM Range: [32151.0, 32217.0]
Apex: 32207.0 Centroid: 31974.47890818438 FWHM Range: [31810.0, 32218.0]
Apex: 32725.0 Centroid: 32726.2381047803 FWHM Range: [32717.0, 32736.0]
Apex: 32761.0 Centroid: 32762.13910467876 FWHM Range: [32747.0, 32778.0]
Apex: 32800.0 Centroid: 32754.30327164134 FWHM Range: [32713.0, 32812.0]
Apex: 32836.0 Centroid: 32765.741009572776 FWHM Range: [32712.0, 32850.0]
Apex: 32893.0 Centroid: 32924.24279873507 FWHM Range: [32881.0, 32975.0]
Apex: 32931.0 Centroid: 32924.68238479144 FWHM Range: [32879.0, 32978.0]
Apex: 32967.0 Centroid: 32841.888963551464 FWHM Range: [32712.0, 33042.0]
Apex: 33011.0 Centroid: 32859.65685022047 FWHM Range: [32711.0, 33109.0]
Apex: 33029.0 Centroid: 32859.44699985565 FWHM Range: [32711.0, 33108.0]
Apex: 33066.0 Centroid: 32859.65685022047 FWHM Range: [32711.0, 33109.0]
Apex: 33099.0 Centroid: 32862.07829020624 FWHM Range: [32710.0, 33124.0]
Apex: 33628.0 Centroid: 33628.75971247903 FWHM Range: [33619.0, 33639.0]
Apex: 33664.0 Centroid: 33646.55248211837 FWHM Range: [33617.0, 33682.0]
Apex: 33702.0 Centroid: 33658.6110746319 FWHM Range: [33615.0, 33718.0]
Apex: 33737.0 Centroid: 33712.992655702394 FWHM Range: [33614.0, 33873.0]
Apex: 33795.0 Centroid: 33713.95362983125 FWHM Range: [33613.0, 33877.0]
Apex: 33811.0 Centroid: 33713.95362983125 FWHM Range: [33613.0, 33877.0]
Apex: 33834.0 Centroid: 33718.6954218247 FWHM Range: [33613.0, 33896.0]
Apex: 33910.0 Centroid: 33735.52919723671 FWHM Range: [33611.0, 33979.0]
Apex: 33969.0 Centroid: 33741.69490383713 FWHM Range: [33610.0, 34020.0]
Apex: 34010.0 Centroid: 33741.99535781033 FWHM Range: [33607.0, 34024.0]
Apex: 34530.0 Centroid: 34530.30350755453 FWHM Range: [34520.0, 34541.0]
Apex: 34568.0 Centroid: 34548.07323270673 FWHM Range: [34518.0, 34583.0]
Apex: 34605.0 Centroid: 34560.03430126617 FWHM Range: [34516.0, 34620.0]
Apex: 34639.0 Centroid: 34571.39066161203 FWHM Range: [34514.0, 34662.0]
Apex: 34714.0 Centroid: 34602.50790443337 FWHM Range: [34513.0, 34769.0]
Apex: 34739.0 Centroid: 34604.68426459892 FWHM Range: [34510.0, 34781.0]
Apex: 35432.0 Centroid: 35449.83829878893 FWHM Range: [35422.0, 35481.0]
Apex: 35469.0 Centroid: 35451.2402212688 FWHM Range: [35420.0, 35487.0]
Apex: 35507.0 Centroid: 35464.36973967245 FWHM Range: [35418.0, 35525.0]
Apex: 35541.0 Centroid: 35476.933778857296 FWHM Range: [35415.0, 35570.0]
Apex: 35585.0 Centroid: 35496.54342663261 FWHM Range: [35412.0, 35650.0]
Plot 2: 0.11s
Plot 4: 0.084s
Saved Figure:  B0.4_7-9min_CombinedScan_Figure1.pdf
Saved Figure:  B0.4_7-9min_CombinedScan_Figure2.pdf
Saved Figure:  B0.4_7-9min_CombinedScan_Figure3.pdf
Saved Figure:  B0.4_7-9min_CombinedScan_Figure4.pdf
Saved Figure:  B0.4_7-9min_CombinedScan_Figure5.pdf
Saved Figure:  B0.4_7-9min_CombinedScan_Figure6.pdf
Saved Figure:  B0.4_7-9min_CombinedScan_Figure1.pdf
Saved Figure:  B0.4_7-9min_CombinedScan_Figure2.pdf
Saved Figure:  B0.4_7-9min_CombinedScan_Figure3.pdf
Saved Figure:  B0.4_7-9min_CombinedScan_Figure4.pdf
Saved Figure:  B0.4_7-9min_CombinedScan_Figure5.pdf
Saved Figure:  B0.4_7-9min_CombinedScan_Figure6.pdf
This is pdfTeX, Version 3.14159265-2.6-1.40.20 (MiKTeX 2.9.7140 64-bit)
entering extended mode
(B0.4_7-9min_CombinedScan_report.tex
LaTeX2e <2018-12-01>
("C:\Program Files\MiKTeX 2.9\tex/latex/base\article.cls"
Document Class: article 2018/09/03 v1.4i Standard LaTeX document class
("C:\Program Files\MiKTeX 2.9\tex/latex/base\size10.clo"))
("C:\Program Files\MiKTeX 2.9\tex/latex/graphics\graphicx.sty"
("C:\Program Files\MiKTeX 2.9\tex/latex/graphics\keyval.sty")
("C:\Program Files\MiKTeX 2.9\tex/latex/graphics\graphics.sty"
("C:\Program Files\MiKTeX 2.9\tex/latex/graphics\trig.sty")
("C:\Program Files\MiKTeX 2.9\tex/latex/graphics-cfg\graphics.cfg")
("C:\Program Files\MiKTeX 2.9\tex/latex/graphics-def\pdftex.def")))
("C:\Program Files\MiKTeX 2.9\tex/latex/amsmath\amsmath.sty"
For additional information on amsmath, use the `?' option.
("C:\Program Files\MiKTeX 2.9\tex/latex/amsmath\amstext.sty"
("C:\Program Files\MiKTeX 2.9\tex/latex/amsmath\amsgen.sty"))
("C:\Program Files\MiKTeX 2.9\tex/latex/amsmath\amsbsy.sty")
("C:\Program Files\MiKTeX 2.9\tex/latex/amsmath\amsopn.sty"))
("C:\Program Files\MiKTeX 2.9\tex/latex/amsfonts\amssymb.sty"
("C:\Program Files\MiKTeX 2.9\tex/latex/amsfonts\amsfonts.sty"))
(C:\Users\Ruisi\AppData\Roaming\MiKTeX\2.9\tex/latex/paracol\paracol.sty)
("C:\Program Files\MiKTeX 2.9\tex/latex/graphics\color.sty"
("C:\Program Files\MiKTeX 2.9\tex/latex/graphics-cfg\color.cfg"))
(C:\Users\Ruisi\AppData\Roaming\MiKTeX\2.9\tex/latex/colortbl\colortbl.sty
("C:\Program Files\MiKTeX 2.9\tex/latex/tools\array.sty"))
("C:\Program Files\MiKTeX 2.9\tex/latex/geometry\geometry.sty"
("C:\Program Files\MiKTeX 2.9\tex/generic/oberdiek\ifpdf.sty")
("C:\Program Files\MiKTeX 2.9\tex/generic/oberdiek\ifvtex.sty")
("C:\Program Files\MiKTeX 2.9\tex/generic/ifxetex\ifxetex.sty")
("C:\Program Files\MiKTeX 2.9\tex/latex/geometry\geometry.cfg"))
(B0.4_7-9min_CombinedScan_report.aux)
(C:\Users\Ruisi\AppData\Roaming\MiKTeX\2.9\tex/context/base/mkii\supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) ("C:\Program Files\MiKTeX 2.9\tex/latex/oberdiek\epstopdf-base.sty"
("C:\Program Files\MiKTeX 2.9\tex/generic/oberdiek\infwarerr.sty")
("C:\Program Files\MiKTeX 2.9\tex/latex/oberdiek\grfext.sty"
("C:\Program Files\MiKTeX 2.9\tex/generic/oberdiek\kvdefinekeys.sty"
("C:\Program Files\MiKTeX 2.9\tex/generic/oberdiek\ltxcmds.sty")))
("C:\Program Files\MiKTeX 2.9\tex/latex/oberdiek\kvoptions.sty"
("C:\Program Files\MiKTeX 2.9\tex/generic/oberdiek\kvsetkeys.sty"
("C:\Program Files\MiKTeX 2.9\tex/generic/oberdiek\etexcmds.sty"
("C:\Program Files\MiKTeX 2.9\tex/generic/oberdiek\ifluatex.sty"))))
("C:\Program Files\MiKTeX 2.9\tex/generic/oberdiek\pdftexcmds.sty"))
*geometry* driver: auto-detecting
*geometry* detected driver: pdftex

Underfull \hbox (badness 10000) in paragraph at lines 13--17


! LaTeX Error: Unknown graphics extension: .4_7-9min_CombinedScan_Figure1.pdf.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...

l.19 ....48]{B0.4_7-9min_CombinedScan_Figure1.pdf}
                                                  \\
!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on B0.4_7-9min_CombinedScan_report.log.
```

**Comment by michaelmarty** [comment]
Glad you are finding it useful!

Yep, it was an issue with parsing file names with a period in the middle. I've got a fix ready. Would you be able to test it for me? I pushed it to the v3 branch. If you use it in the compiled version, send me an email, and I can send you a link directly. I don't want to push a new master release until we've tested it a bit. Thanks, MTM

**Comment by michaelmarty** [comment]
Fixed with Version 4.0.2
